app/src/main/java/de/hhn/aib/labsw/blackmirror/controller/API/Websockets/MirrorApiWebsockets.kt

```diff
@@ -0,0 +1,118 @@
+package de.hhn.aib.labsw.blackmirror.controller.API.Websockets
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import okhttp3.*
+import java.io.IOException
+
+class MirrorApiWebsockets : WebSocketListener(),MirrorApi {
+    var sessions: ArrayList<WebSocket> = ArrayList()
+    private var mapper: ObjectMapper = ObjectMapper()
+    var listeners: HashMap<String, ArrayList<ApiListener>?> =
```

Comment:
Why is the list of listeners for a specific topic nullable? If there are no listeners, there should simply be no entry in the hash map!
As a general rule of thumb, u probably should never need to make a collection nullable i think